src/lib/encoder.ts

```diff
@@ -132,6 +132,54 @@ const valueTypeEncodingMap = {
     encode: (value: string[]) => abiCoder.encodeParameter('bytes[]', value),
     decode: (value: string) => abiCoder.decodeParameter('bytes[]', value),
   },
+  'bytes[CompactBytesArray]': {
+    encode: (value: string[]) => encodeCompactBytesArray(value),
+    decode: (value: string) => decodeCompactBytesArray(value),
+  },
+};
+
+const encodeCompactBytesArray = (values: string[]): string => {
+  let compactBytesArray = '0x';
+  for (let i = 0; i < values.length; i++) {
+    if (!isHex(values[i]))
+      throw new Error(`Couldn't encode, value at index ${i} is not hex`);
+
+    if (values[i].length > 65_535 * 2 + 2)
+      throw new Error(
+        `Couldn't encode bytes[CompactBytesArray], value at index ${i} exceeds 65_535 bytes`,
+      );
+
+    const numberOfBytes = (values[i].length - 2) / 2;
+    let hexNumber = padLeft(numberToHex(numberOfBytes), 4);
+
+    compactBytesArray += hexNumber.substring(2) + values[i].substring(2);
+  }
+
+  return compactBytesArray;
+};
+
+const decodeCompactBytesArray = (compactBytesArray: string): string[] => {
+  if (!isHex(compactBytesArray))
```

Comment:
does `isHex` returns true on empty string? because at the SC level we consider 0x to be a valid CBA and we may have keys with empty CBA ? could not see a test for that)

Reply:
@skimaharvey added test for this case scenario